server/src/resolvers/trainingResolver.ts

```diff
@@ -69,14 +71,66 @@ const TrainingResolvers = {
     submitModule: async (
       _parent: any,
       args: { moduleID: number; answerSheet: AnswerInput[] },
-      { ifAllowed }: ApolloContext
+      { ifAllowed, user }: ApolloContext
```

Comment:
I don't think we need `user` here. We should use the one in `ifAllowed`'s callback arguments, since that one is guaranteed to exist.